ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -352,19 +355,15 @@ Joiner getJoiner(final DataSource.DataSourceType leftType,
       );
     }
 
-    @SuppressWarnings("OptionalGetWithoutIsPresent")
-    Field getJoinKey(final String alias, final KeyField keyField) {
-      final KeyField keyFieldWithAlias = KeyField.of(
-          keyField.name().map(name -> SchemaUtil.buildAliasedFieldName(alias, name)),
+    static KeyField getJoinKey(final String alias, final KeyField keyField) {
+      final String latest = keyField.name()
+          .map(name -> SchemaUtil.buildAliasedFieldName(alias, name))
+          .orElse(SchemaUtil.buildAliasedFieldName(alias, SchemaUtil.ROWKEY_NAME));
```

Comment:
oooh! what if we have a `KeyField#aliased(String)` that turns both the elements into aliases? Then this code becomes so simple:
```java
final String latest = keyField.name().orElse(ROWKEY_NAME);
return KeyField.of(Optional.of(latest), keyField.legacy()).aliased(alias);
```

Reply:
I did think about that, but thought its potentially putting too much logic into `KeyField`.  I'll take another look.

Reply:
Done.